stix_shifter_modules/gcp_chronicle/stix_transmission/results_connector.py

```diff
@@ -0,0 +1,374 @@
+                                response_dict['message'] = response_text['error'].get('message')
+                                ErrorResponder.fill_error(return_obj, response_dict, ['message'],
+                                                          connector=self.connector)
+                                return_obj = self.invalid_response(return_obj, response_dict,
+                                                                   next_response[0].status, response_text)
+                                break
+                        else:
+                            break
+                return_obj['success'] = True
+                if result:
+                    final_result = ResultsConnector.get_results_data(result)
+                    return_obj['data'] = final_result[int(offset):(int(offset) + int(length))] if final_result else []
+                else:
+                    return_obj['data'] = []
+                # delete the error details if partial results are fetched
+                if 'code' in return_obj and 'error' in return_obj and return_obj['data']:
+                    del return_obj['code']
+                    del return_obj['error']
+            else:
+                return_obj = self.invalid_response(return_obj, response_dict, response_wrapper[0].status,
+                                                   response_text)
+
+        except ServerNotFoundError:
+            response_dict['code'] = 1010
+            response_dict['message'] = "Invalid Host"
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+
+        except RefreshError:
+            response_dict['code'] = 1015
+            response_dict['message'] = "Invalid Client Email"
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+
+        except ValueError as r_ex:
+            if 'Could not deserialize key data' in str(r_ex):
+                response_dict['message'] = r_ex
+                response_dict['code'] = 1015
+            else:
+                response_dict['message'] = f'cannot parse {r_ex}'
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+
+        except Exception as exp:
+            if "timed out" in str(exp):
+                response_dict['code'] = 120
+                response_dict['message'] = str(exp)
+            else:
+                response_dict['message'] = exp
+            self.logger.error('error when getting search results: %s', exp)
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+
+        finally:
+            if 'code' in response_dict:
+                if response_dict['code'] not in (1010, 1015):
+                    self.api_client.delete_search(search_id)
+            else:
+                self.api_client.delete_search(search_id)
```

Comment:
why are you deleting the search after every results call? I thought we decided to delete when the actual delete operation is called. 

Also, Results call returns error if the search is deleted. this happens when we run the results call again with the same search ID to retrieve the remaining results.

Reply:
Delete API call is added in order to avoid the overload of the chronicle database. Even if the users doesn't have permission to make the deletion API call, the code will return the results without exception.
The Deletion API call will be done only when all the required records are fetched or when next_page_token is not available(Detailed information has been sent in email).